dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -334,11 +362,14 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
                     winrt::guid registeredClsid{ GUID_NULL };
                     if (AppModel::Identity::IsPackagedProcess())
                     {
-                        registeredClsid = PushNotificationHelpers::GetComRegistrationFromRegistry(expectedAppServerArgs.data());
+                        registeredClsid = PushNotificationHelpers::GetComRegistrationFromRegistry(c_expectedAppServerArgs.data());
                     }
                     else
                     { 
-                        registeredClsid = RegisterComActivatorGuidAndAssets();
+                        std::wstring registeredClsidString;
+                        THROW_IF_FAILED(GetActivatorGuid(registeredClsidString));
+
+                        registeredClsid = winrt::guid(registeredClsidString.substr(1, registeredClsidString.size() - 2));
```

Comment:
General comment. I see a lot of string to guid conversions and vice versa. There are also operations to remove braces from the string. Can we simply store the guid as winrt::guid and use that everywhere instead of the different conversions we have all over the place?

Reply:
The reason for some of these conversions is to read/write to the registry. If there is a conversion from winrt::guid<->byte, I believe we can completely remove the std::wstring<->winrt::guid conversions.

Reply:
Actually, we should probably leave this to the compiler. Based on what I am reading here: https://en.cppreference.com/w/cpp/language/copy_elision - the compiler tries to do a return value optimization first i.e.-"objects are constructed directly into the storage where they would otherwise be copied/moved to." - if it fails then it would switch to move and then finally to copy. Trying to std::move would end up counterintuitive to what is being achieved. I would be avoiding this in the PR. Let me know if you otherwise. Thanks.

Reply:
I created an internal task for this to track. Thanks.